estate/models/estate_property_offer.py

```diff
@@ -0,0 +1,63 @@
+from dateutil.relativedelta import relativedelta
+from odoo import api, models, fields
+from odoo.exceptions import UserError
+
+
+class EstatePropertyOffer(models.Model):
+    _name = "estate.property.offer"
+    _description = "estate property offer"
+    _order = "price desc"
+
+    price = fields.Float()
+    status = fields.Selection(
+        selection=[('accepted', 'Accepted'), ('refused', 'Refused')],
+        copy=False
+    )
+    partner_id = fields.Many2one('res.partner', string="Partner", required=True)
+    property_id = fields.Many2one('estate.property', string="Property", required=True, ondelete="cascade")
+    validity = fields.Integer(default=7)
+    date_deadline = fields.Date(compute="_compute_date_deadline", inverse="_inverse_date_deadline")
+    property_state = fields.Selection(related="property_id.state")
+    property_type_id = fields.Many2one(related="property_id.property_type_id", store=True)
+
+    _positive_offer_price = models.Constraint(
+        'CHECK(price > 0)',
+        'The Offer Price must be strictly positive'
+    )
+
+    @api.depends("validity", "create_date")
+    def _compute_date_deadline(self):
+        for offer in self:
+            offer.date_deadline = (offer.create_date or fields.Date.today()) + relativedelta(days=+offer.validity)
+
+    def _inverse_date_deadline(self):
+        for offer in self:
+            offer.validity = (offer.date_deadline - (offer.create_date or fields.Date.today()).date()).days
+
+    @api.model
+    def create(self, vals):
+        for offer in vals:
+            prop = self.env['estate.property'].browse(offer['property_id'])
+            prop.state = "offer_received"
+            if offer['price'] < prop.best_price:
```

Comment:
You shouldn't do sql request inside a loop 😃. It will do a lot of requests to your db as the create method can be used to create multiple records. Get first all the needed ids and then make a request with all of them with_prefetch(), which will allow you to get records by their id with browse inside the loop without doing any requests to the data base.  Here also you can see that I used EstateProperties instead of estate_properties, that's an Odoo convention. When a variable represent a model we have to use the same convention than for the name of a python class.
```suggestion
   def create(self, vals_list):
       EstateProperties = self.env["estate.property"].with_prefetch(
            [vals["property_id"] for vals in vals_list]
        )
        for offer in vals_list:
            property = EstateProperties.browse(offer["property_id"])
```